app/javascript/react/containers/PodcastShowContainer.js

```diff
@@ -23,18 +29,63 @@ class PodcastShowContainer extends Component {
       })
       .then(response => response.json())
       .then(podcast => {
-        this.setState({ podcast: podcast["podcasts"][0] });
+        this.setState({
+          name: podcast["podcast"]["name"],
+          description: podcast["podcast"]["description"],
+          publisher: podcast["podcast"]["publisher"],
+          link: podcast["podcast"]["link"],
+          reviews: podcast["podcast"]["reviews"]
+        });
+      })
+      .catch(error => console.log(`Error in fetch: ${error.message}`));
+  }
+
+  addNewReview(formPayload) {
+    fetch("/api/v1/reviews", {
+      credentials: "same-origin",
+      method: "POST",
+      body: JSON.stringify(formPayload),
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json"
+      }
+    })
+      .then(response => {
+        if (response.ok) {
+          return response;
+        } else {
+          let errorMessage = `${response.status} (${response.statusText})`,
+            error = new Error(errorMessage);
+          throw error;
+        }
+      })
+      .then(response => response.json())
+      .then(newReview => {
```

Comment:
OH I forgot to say this earlier. The one thing I want you to add here is error handling! You can get into this block whether or not you send back a successful review (try not filling out a field and see if you hit a debugger here). 

Ideally, my suggestion would be to validate BEFORE doing the post, see here: https://learn.launchacademy.com/teams/boston-23/curricula/on-campus-boston-23/lesson_groups/week_4:_advanced_react/lessons/react-form-validations

But another option would be to add an `if` statement here, checking `if(newReview.errors)`.

I'm willing to merge this without the error handling since your existing functionality is so well-tested, but please consider adding this!

Reply:
will do!